### PR TITLE
Upgrade bulk editor modules, #9219

### PR DESCRIPTION
**Pull request by njkim**
### Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description of Change

First round of improving the bulk editor.
The string bulk editor is divided into small functions.
Also, the preview were added, case insensitive string replace, combine trim with other functions were added.

Further improvement will be addressed in other tickets. #9366, #9367, #9368, #9369

### Issues Solved

#9219, #9220, #9222

### Checklist

-   [ ] Unit tests pass locally with my changes
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] I have added necessary documentation (if appropriate)

#### Ticket Background
*   Sponsored by: 
*   Found by: @ 
*   Tested by: @ 
*   Designed by: @

### CODE REVIEW

File: arches/app/templates/views/components/etl_modules/base-editor.htm
Comment (apeters) [inline arches/app/templates/views/components/etl_modules/base-editor.htm:1]: This file should probably be renamed too.  Something corresponds to it's .js file `base-bulk-string-editor.htm`

**Review by apeters — CHANGES_REQUESTED** [review]
I think the directory structure needs to be reviewed and potentially refactored.  Let's discuss.

File: arches/app/etl_modules/base_data_editor.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 logger = logging.getLogger(__name__)
 
 
-class BulkDataEditor(BaseImportModule):
+class BaseDataEditor(BaseImportModule):
```
[the remark this answers is not in the document]
Comment (njkim) [reply arches/app/etl_modules/base_data_editor.py:18]: The file names and class names are updated and the migration is separated and updated accordingly.
The update are going to be reflected in the pr #9383

**Comment by njkim** [comment]
The branch is merged into https://github.com/archesproject/arches/pull/9383

**Pull request closed without merging**